### ISSUE TITLE
Long time required in solving

### ISSUE THREAD

**Issue by SoumiDas** [cvxgrp/cvxpylayers#86]
Hi,
I was trying to solve an optimisation problem and witnessed that using cvxpy, it takes approximately ~0.4 seconds while using cvxpylayers, it keeps varying with the minimum being 1.5+ seconds. Can you please let me know if I am doing any mistake or any reason why it is so? The code is attached underneath:

`import numpy as np
import cvxpy as cp
import time
import torch
from cvxpylayers.torch import CvxpyLayer

device = 'cuda' if torch.cuda.is_available() else 'cpu'
m =100
n =60
m1 = m - n
fraction = 0.2  # fraction of original points to be selected
print("Existing and Incoming set count "+str(m1)+","+str(n))

def subsetLayer(D_in, D_exin):
  
  finc = cp.Parameter((D_in, D_in))
  fex = cp.Parameter((D_in, D_exin))
 
  Zn = cp.Variable((D_in, D_in))
  Zo = cp.Variable((D_in, D_exin))
  
  obj = cp.sum(cp.multiply(fex, Zo))+cp.sum(cp.multiply(finc, Zn))
  
  M = cp.sum(Zn, axis=1) + cp.sum(Zo,axis=1)
  N_p = cp.norm(Zn, p=1, axis=0)
  est_size = cp.norm(N_p, p=1)
  constraints = [0 <= Zo, Zo <= 1, 0 <= Zn, Zn <= 1, M == 1, est_size <= fraction*D_in]
  objective = cp.Minimize(obj)
  prob = cp.Problem(objective, constraints)
  print("Solving using cvxpylayers")
  layer = CvxpyLayer(prob, parameters=[finc, fex], variables=[Zn])

  return layer

D_in = 60
D_exin = 40

layer = subsetLayer(D_in,D_exin)

Do = np.random.randn(n, m1)
Dn = np.random.randn(n,n)

Doc = torch.from_numpy(Do).float().to(device)
Dnc = torch.from_numpy(Dn).float().to(device)

cvstart = time.time()
zv = layer(Dnc, Doc)[0]
print("Using CVXPYLayers: Existing "+str(D_exin)+" Incoming "+str(D_in))
print(time.time() - cvstart)

############## CVXPY####################

cvstart = time.time()
Zn = cp.Variable((n, n))
Zo = cp.Variable((n, m1)) 

obj = cp.sum(cp.multiply(Do, Zo))+cp.sum(cp.multiply(Dn, Zn))
M = cp.sum(Zn, axis=1) + cp.sum(Zo,axis=1)
N_p = cp.norm(Zn, p=1, axis=0) 
est_size = cp.norm(N_p, p=1)
constraints = [0 <= Zo, Zo <= 1, 0 <= Zn, Zn <= 1, M == 1, est_size <= fraction*n]

objective = cp.Minimize(obj)
prob = cp.Problem(objective, constraints)
print("Solving using cvxpy")
result = prob.solve()
print("Using CVXPY: Existing "+str(m1)+" Incoming "+str(n))
print(time.time()-cvstart)`

Thanks!

**Comment by sbarratt** [comment]
You are seeing different solve times because two different numerical
solvers are being used. CVXPYlayers uses the solver "SCS" by default, but
CVXPY uses the solver "ECOS" by default for problems like these. At this
point in time, SCS and ECOS are the only supported solvers by CVXPYlayers.
To change the solver to ECOS, you would change the line where you call
`layer` to:
```
zv = layer(Dnc, Doc, solver_args={"solve_method": "ECOS"})[0]
```

Changing this results in a 0.259 solve time for CVXPYlayers and a 0.347
solve time for CVXPY, which are much more similar.

Note that if your problem includes exponential or PSD cones, and you want
to use CVXPYlayers, you must use SCS.
